### PR TITLE
Rewrite ReconInput generator

### LINKED ISSUE #100: Modify automated newick trees generation to use two/tree files instead of one
Will disable the current unittest that uses automated tree generation

### PR DESCRIPTION
**Pull request by ssantichaivekin**
Rewrite the generator that generates all or random `ReconInput` objects. Add saving functionality to `ReconInput` so that one can save generated inputs.

Resolves #100 .

### CODE REVIEW

File: empress/miscs/input_generator.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    """
+    We don't actually generate all possible trees in this
+    format. In the code, you will see that we only generate a right-heavy
+    tree, meaning that the right subtree has equal or greater number of
+    nodes than the left subtree. We do this to reduce the number of trees
+    that are isomorphic to one another.
+    """
+    tree_dict = {}
+    if n_leaves == 1:
+        tree_dict[top_name] = ('Top', prefix + '0', None, None)
```
Comment (JingyiRose) [inline empress/miscs/input_generator.py:44]: can we handle this case in generate_all_trees_helper by checking whether n_leaves is 0?

Comment (ssantichaivekin) [reply empress/miscs/input_generator.py:44]: I'm not sure I understand your comment. We only call the `_generate_all_trees_helper` if we have at least two leaves. Are you saying that even if we have one leaf, we should call both `_generate_all_trees_helper` on the left subtree and on the right subtree even if there is only one subtree? Would you mind clarifying how this should be modified?

**Comment by ssantichaivekin** [comment]
I'm thinking that `miscs` could contain things that are not relevant to the main workflow of empress. I'm thinking of a name that represents "other things that are not very important".

File: empress/miscs/input_generator.py (hunk 1 above)
Comment (JingyiRose) [reply empress/miscs/input_generator.py:44]: yeah, your understanding is correct. This is only a minor suggestion, the code works fine either way~

Comment (ssantichaivekin) [reply empress/miscs/input_generator.py:44]: I still don't understand your comment, and it does not seem that you are planning to explain it, so I will just merge.

**Pull request merged**